l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -43,6 +44,13 @@ def check_normalized_string(value):
     return normalized
 
 
+def normalize_encode(value, form='NFKD', encoding='ascii', errors='replace'):
+    res = ''
+    if value:
+        res = unicodedata.normalize(form, value).encode(encoding=encoding, errors=errors)
```

Comment:
@vincenzoterzulli: l'istruzione corretta per il problema che hai indicato è la seguente.

```suggestion
        res = value.encode('latin1', 'replace').decode('latin1')
```

---

Cercherò di essere molto semplice e conciso (anche se, sicuramente, non ci riuscirò) nell'illustrarti cos'ha che non va la tua soluzione proposta.

Il carattere da te indicato (`ß`) in questo esempio **È**, a tutti gli effetti, un carattere **VALIDO** e **ESISTENTE** nella codifica `latin1`.  
Per questo motivo, quando si esegue su tale stringa `.encode('latin1', 'ignore')` e poi, successivamente, `.decode('latin')`, è normale che il carattere di partenza non venga rimosso ma, bensì, mantenuto.  

Questo, appena descritto, deve essere il comportamento corretto da implementare in questa PR.

---

Facendo riferimento alla [documentazione ufficiale](https://www.agenziaentrate.gov.it/portale/it/web/guest/schede/comunicazioni/dati-fatture-c.d.-nuovo-spesometro-/specifiche-tecniche-dati-fatture-c.d.-nuovo-spesometro-?page=schedecomunicazioni) dell'AdE e al [file di definizione](https://www.agenziaentrate.gov.it/portale/documents/20143/296761/Provvedimento+05022018+dati+fattura_DatiFattura_v2.1.xsd/f471c8ed-ea26-b0b0-4e9f-09e0633b1f03) in formato XSD, ti riporto la definizione dei tipi dei campi (comprese tra le righe 357 e 385):
```xsd
  <xs:simpleType name="String10Type">
    <xs:restriction base="xs:normalizedString">
      <xs:pattern value="(\p{IsBasicLatin}{1,10})" />
    </xs:restriction>
  </xs:simpleType>
  
  <xs:simpleType name="String18Type">
    <xs:restriction base="xs:normalizedString">
      <xs:pattern value="(\p{IsBasicLatin}{1,18})" />
    </xs:restriction>
  </xs:simpleType>
  
  <xs:simpleType name="String20Type">
    <xs:restriction base="xs:normalizedString">
      <xs:pattern value="(\p{IsBasicLatin}{1,20})" />
    </xs:restriction>
  </xs:simpleType>
    
  <xs:simpleType name="String60LatinType">
    <xs:restriction base="xs:normalizedString">
      <xs:pattern value="[\p{IsBasicLatin}\p{IsLatin-1Supplement}]{1,60}" />
    </xs:restriction>
  </xs:simpleType>

  <xs:simpleType name="String80LatinType">
    <xs:restriction base="xs:normalizedString">
      <xs:pattern value="[\p{IsBasicLatin}\p{IsLatin-1Supplement}]{1,80}" />
    </xs:restriction>
  </xs:simpleType>
```

In queste definizioni, vengono utilizzati due termini per definire i caratteri validi che, ogni tipo, può accettare:
*  [Basic Latin](https://it.wikipedia.org/wiki/Basic_Latin): fondamentalmente il set di caratteri valido per la codifica `ascii`.
*  [Latin-1 Supplement](https://it.wikipedia.org/wiki/Latin-1_Supplement): tutti quei caratteri aggiuntivi che, insieme al set disponibile per la codifica `ascii` compongono un set valido di caratteri per la codifica `latin1` tra cui, puoi vedere elencato anche il nostro carattere di partenza `ß` (che, quindi, a maggior ragione non deve essere rimosso).

---

Ti faccio, quindi, una domanda...  
Immagino tu abbia aperto questa PR proprio dopo esserti trovato rigettato il file per un errore di codifica.

Ci sai, eventualmente, indicare con precisione quale dei campi ha generato tale errore?
È possibile che l'errore di rigetto del file possa essere stato causato da un altro carattere in qualche altro campo?

Reply:
Ciao @Byloth,

* l'ADE richiede il latin altrimenti rigetta il file (come ho scritto nella issue) e non permette di procedere. Quindi rigetta per il formato errato e non perché un campo contiene simboli non riconosciuti.
 
![image](https://user-images.githubusercontent.com/16492632/67585699-62e30400-f750-11e9-8735-93360f640cd9.png)


* forzo la conversione ad ASCII invece che a LATIN per evitare a prescindere possibili errori di interpretazione (vedi a destra nell'immagine) ben consapevole che alcuni simboli ci sono in LATIN
![image](https://user-images.githubusercontent.com/16492632/67585212-645ffc80-f74f-11e9-8fb8-b5ed07049c53.png)


I campi coinvolti sono per lo più Indirizzo e Denominazione, ma ovviamente è possibile che si verifichi su tutti i campi testuali di tipo `String60LatinType` e `String80LatinType` , ovvero anche Nome, Cognome e Comune
![image](https://user-images.githubusercontent.com/16492632/67596430-fb857e00-f768-11e9-8679-9c6bae16a7d5.png)

Ad ogni modo tra lunedì' e martedì farò delle prove di trasmissione all'ADE dei file XML estratti con entrambi i metodi. Vi faccio sapere i risultati.